concepts/50 React Components/40 Component Configuration Syntax/55 Markup Customization/0 Markup Customization.md

```diff
@@ -1,5 +1,7 @@
-Templates allow you to customize UI components. In the DevExtreme API, template properties end with `Template`: `itemTemplate`, `groupTemplate`, `contentTemplate`. When you specify them in React, replace `Template` in the name with `Render` or `Component`, depending on whether the template is a rendering function or custom component. For example, instead of `itemTemplate`, use `itemRender` or `itemComponent`. If a component does not reinitiate a template when the component rerenders, you should use the `itemComponent` property. Otherwise, we recommend that you use the `itemRender` property. 
+You can define custom content in DevExtreme components. We supply 2 types of properties for this task: ones that end with *'render'* (for example, [cellRender](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/#cellRender)), and ones that end with *'component'* (for example, [cellComponent](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/#cellComponent)).
 
-If the property is called `template`, without any prefix, (in the [Button](/api-reference/10%20UI%20Components/dxButton/1%20Configuration/template.md '/Documentation/ApiReference/UI_Components/dxButton/Configuration/#template'), [Drawer](/api-reference/10%20UI%20Components/dxDrawer/1%20Configuration/template.md '/Documentation/ApiReference/UI_Components/dxDrawer/Configuration/#template'), and other UI components), use the `render` or `component` property instead.
+The *'render'* properties allow you to employ a rendering function. Whatever you return with this function is rendered in a virtual DOM. The rendering function is not a component: it lacks state and cannot utilize hooks. However, you can create the rendering function inside your component, enclosed within the [useCallback](https://react.dev/reference/react/useCallback). You can pass local variables as dependencies, allowing you to use them in your template. Refer to the following article to see an example: [Using a Rendering Function](/Documentation/Guide/React_Components/Component_Configuration_Syntax/#Markup_Customization/Using_a_Rendering_Function).
 
-[tags] template
+With *'component'* properties, you work with a single component as a template. This template component is rendered in the virtual DOM. Since it is rendered as a component, it can have its internal state and use hooks inside, which is not possible with render functions. Refer to the following article to see an example: [Using a Custom Component](/Documentation/Guide/React_Components/Component_Configuration_Syntax/#Markup_Customization/Using_a_Custom_Component).
```

Comment:
I'm trying to navigate between the words "component" that may mean a property, a React component concept, or a DevExpress component. 

Also, you were calling them "rendering functions" in the previous paragraph, but this one says "render functions". I'm changing it for consistency. 

```suggestion
*'Component'* properties use React components to specify templates. A DevExpress control renders those templates to the virtual DOM. Since a template is a component, it can have its internal state and use hooks inside (which is not possible with rendering functions). The following article contains an example: [Using a Custom Component](/Documentation/Guide/React_Components/Component_Configuration_Syntax/#Markup_Customization/Using_a_Custom_Component).
```